server/src/main/java/org/apache/druid/server/coordinator/duty/KillUnusedSegments.java

```diff
@@ -140,6 +162,12 @@ private void killUnusedSegments(Collection<String> dataSourcesToKill)
           break;
         }
       }
+
+      if (submittedTasks >= availableKillTaskSlots) {
+        log.info("Reached kill task slot limit with pending unused segments to kill. Will resume "
```

Comment:
Might as well include the task slot limit and how many tasks are submitted, something like:
```suggestion
        log.info("Submitted [%d] kill tasks and reached kill task slot limit [%d]. Will resume "
```